utils/frame/benchmarking-cli/src/pallet/writer.rs

```diff
@@ -397,38 +447,145 @@ pub(crate) fn add_storage_comments(
+				(true, true) => continue,
+				(false, true) => {
+					// track newly identified key
+					identified_key.insert(key.clone());
 				},
-				None => {
-					let comment = format!(
-						"Storage: unknown [0x{}] (r:{} w:{})",
-						HexDisplay::from(key),
-						reads,
-						writes,
-					);
-					comments.push(comment)
+				(false, false) => {
+					// track newly identified key and prefix
+					identified_key.insert(key.clone());
+					identified_prefix.insert(prefix.clone());
 				},
+				// Not possible. If the key is known, the prefix is too.
+				(true, false) => unreachable!(),
+			}
+
+			// For any new prefix, we should write some comment about the number of reads and
+			// writes.
+			if !is_prefix_identified {
+				match storage_info_map.get(&prefix) {
+					Some(key_info) => {
+						let comment = format!(
+							"Storage: {} {} (r:{} w:{})",
+							String::from_utf8(key_info.pallet_name.clone())
+								.expect("encoded from string"),
+							String::from_utf8(key_info.storage_name.clone())
+								.expect("encoded from string"),
+							reads,
+							writes,
+						);
+						comments.push(comment)
+					},
+					None => {
+						let comment = format!(
+							"Storage: unknown [0x{}] (r:{} w:{})",
+							HexDisplay::from(key),
+							reads,
+							writes,
+						);
+						comments.push(comment)
+					},
+				}
+			}
+
+			// For any new key, we should add the PoV impact.
+			if !is_key_identified {
+				match storage_info_map.get(&prefix) {
+					Some(key_info) => {
+						match worst_case_pov(
+							key_info.max_values,
+							key_info.max_size,
+							!is_prefix_identified,
+							worst_case_map_values,
+						) {
+							Some(new_pov) => {
+								max_pov += new_pov;
+								let comment = format!(
+									"Proof: {} {} (max_values: {:?}, max_size: {:?}, added: {})",
+									String::from_utf8(key_info.pallet_name.clone())
+										.expect("encoded from string"),
+									String::from_utf8(key_info.storage_name.clone())
+										.expect("encoded from string"),
+									key_info.max_values,
+									key_info.max_size,
+									new_pov,
+								);
+								comments.push(comment)
+							},
+							None => {
+								let comment = format!(
+									"Proof Skipped: {} {} (max_values: {:?}, max_size: {:?})",
+									String::from_utf8(key_info.pallet_name.clone())
+										.expect("encoded from string"),
+									String::from_utf8(key_info.storage_name.clone())
+										.expect("encoded from string"),
+									key_info.max_values,
+									key_info.max_size,
+								);
+								comments.push(comment)
+							},
+						}
+					},
+					None => {
+						let comment = format!(
+							"Proof Skipped: unknown [0x{}] (r:{} w:{})",
+							HexDisplay::from(key),
+							reads,
+							writes,
+						);
+						comments.push(comment)
+					},
+				}
 			}
 		}
 	}
+
+	max_pov
+}
+
+// Given the max values and max size of some storage item, calculate the worst
+// case PoV.
+fn worst_case_pov(
```

Comment:
would like to see more documentation on this as the logic is not trivial to follow.

Reply:
a basic example walk-through might be good.